database/db.go

```diff
@@ -51,19 +52,32 @@ type dbLogger struct {
 	logger logging.Logger
 }
 
+// context key for query timing context
+var dlTimingKey struct{} = struct{}{}
+
 // BeforeQuery is a function that will be invoked
 // before any database query is run with the query to run.
-func (d dbLogger) BeforeQuery(q *pg.QueryEvent) {}
+func (d dbLogger) BeforeQuery(q *pg.QueryEvent) {
+	if q.Ctx == nil {
+		q.Ctx = context.Background()
+	}
+	q.Ctx = context.WithValue(q.Ctx, dlTimingKey, time.Now())
+}
 
 // AfterQuery is a function that will be executed
 // after any database query is run with the query ran.
 func (d dbLogger) AfterQuery(q *pg.QueryEvent) {
 	query, err := q.FormattedQuery()
 	if err != nil {
-		d.logger.Printf("error %s executing query\n%+v ", err, query)
+		d.logger.Errorf("error %q executing query:\n%+v ", err, query)
```

Comment:
error formatting query? (go-pg docs are confusing but it looks like this formattedQuery isn't what actually runs the query